### ISSUE TITLE
proposal: `specify_parameter_names`

### ISSUE THREAD

**Issue by goderbauer** [dart-lang/linter#4102]
## Description

Specify names for the parameters of Function types. 

## Details

Dart allows omitting  the name of parameters in function types:

```dart
class LayoutBuilder {
  LayoutBuilder({this.builder});

  final Widget Function(BuildContext, ConstraintType) builder;
}
```

While that is legal, it degrades the experience of anyone using this widget with code completion in IDEs:

![image](https://user-images.githubusercontent.com/1227763/222303636-7b1ce19b-3a3f-46c6-b2c1-22fdb06045c6.png)

Since no names were given, unhelpful names are generated. This can be avoided by giving the parameters proper names:

```dart
class LayoutBuilder {
  LayoutBuilder({this.builder});

  final Widget Function(BuildContext context, ConstraintType constraints) builder;
}
```

The proposed lint would enforce that function types must always contain names for the parameters to improve the experience for anyone using IDE and code completion.

See also https://github.com/flutter/flutter/pull/119877.

## Kind

Maybe style? It's kinda a usability thing when people work with IDEs, though.

## Bad Examples

```dart
Widget Function(BuildContext, ConstraintType) builder;
typedef BuilderCallback = Widget Function(BuildContext, ConstraintType);
```

## Good Examples 

```dart
Widget Function(BuildContext context, ConstraintType constraints) builder;
typedef BuilderCallback = Widget Function(BuildContext context, ConstraintType constraints);
```

**Comment by goderbauer** [comment]
cc @bebe0612
